astropy/utils/data.py

```diff
@@ -232,6 +233,7 @@ def get_readable_fileobj(name_or_obj, encoding=None, cache=False,
         try:
             import bz2
         except ImportError:
+            fileobj.close()
```

Comment:
*Something* is needed if a string is passed in, because at this point we already have an open file handle that we're not passing back to the caller.